src/index.ts

```diff
@@ -64,18 +64,61 @@ export async function getRate(sourceToken: Token, targetToken: Token, amount: st
     return await getRateByPath(paths, amount);
 }
 
-export async function getAllPaths(sourceToken: Token, targetToken: Token) {
+export async function getAllPathsAndRates(sourceToken: Token, targetToken: Token, amount: string = '1') {
     if (sourceToken.blockchainType == 'ethereum' && targetToken.blockchainType == 'ethereum')
-        return await ethereumGetAllPaths(sourceToken.blockchainId, targetToken.blockchainId);
+        return await ethGetAllPathsAndRates(sourceToken.blockchainId, targetToken.blockchainId, amount);
     throw new Error(sourceToken.blockchainType + ' blockchain to ' + targetToken.blockchainType + ' blockchain not supported');
 }
 
+function getEthShortestPath(paths: string[][], rates: string[]): string[] {
+    let index = 0;
+    for (let i = 1; i < paths.length; i++) {
+        if (betterPath(paths, index, i) || (equalPath(paths, index, i) && betterRate(rates, index, i)))
+            index = i;
+    }
+    return paths[index];
+}
+
+function getEthCheapestPath(paths: string[][], rates: string[]): string[] {
+    let index = 0;
+    for (let i = 1; i < rates.length; i++) {
+        if (betterRate(rates, index, i) || (equalRate(rates, index, i) && betterPath(paths, index, i)))
+            index = i;
+    }
+    return paths[index];
+}
+
+function betterPath(paths: string[][], index1: number, index2: number): boolean {
```

Comment:
Seems like 'shortest', not 'better'

Reply:
`betterPath` / `betterRate`.
The meaning for "better" is implementation-specific.
That's my opinion of course (I actually renamed them from `shorterPath`  and `cheaperRate` to the current version).

Reply:
It doesn't tell the user much, I'd rather stick to cold, self explanatory definitions

Reply:
@yudilevi : It's not for the user (internal functions).

Reply:
users are also devs that want to contribute to the repo and regardless, internal functions should be as readable as external ones

Reply:
OK, but let's do only the musts on this branch, and the rest on the `redesign_and_cleanups` branch. Otherwise I'm spending time doing the same work twice.
And this is not a must, right?

Reply:
sure